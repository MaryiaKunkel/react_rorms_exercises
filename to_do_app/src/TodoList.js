import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import { v4 as uuidv4 } from "uuid";

function TodoList() {
  const [todos, setTodos] = useState([]);
  const addTodo = (newTodo) => {
    setTodos((todos) => [...todos, { ...newTodo, id: uuidv4() }]);
  };

  const deleteTodo = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      <ul>
        {todos.map(({ id, text }) => (
          <li>
            <Todo id={id} text={text} onDelete={deleteTodo} />
          </li>
        ))}
      </ul>
    </div>
  );
}

export default TodoList;
