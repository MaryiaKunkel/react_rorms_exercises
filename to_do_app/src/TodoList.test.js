import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new todo", function () {
  const { queryByText, getByPlaceholderText, getByTestId } = render(
    <TodoList />
  );
  expect(queryByText("X")).not.toBeInTheDocument();

  const input = getByPlaceholderText("type in your task");

  fireEvent.change(input, { target: { value: "do dishes" } });

  const button = queryByText("Add task");
  fireEvent.click(button);

  const task = getByTestId("task");
  expect(task).toBeInTheDocument();
  expect(queryByText("X")).toBeInTheDocument();
});
