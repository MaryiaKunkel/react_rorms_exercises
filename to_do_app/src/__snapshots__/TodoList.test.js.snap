// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        id="text"
        name="text"
        placeholder="type in your task"
        type="text"
        value=""
      />
      <button>
        Add task
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
